homeassistant/components/octoprint/sensor.py

```diff
@@ -211,16 +211,23 @@ def native_value(self):
 
         for temp in printer.temperatures:
             if temp.name == self._api_tool:
-                return round(
+                val = (
                     temp.actual_temp
                     if self._temp_type == "actual"
-                    else temp.target_temp,
-                    2,
+                    else temp.target_temp
                 )
+                if not val:
+                    return None
+
+                return round(val, 2)
 
         return None
 
     @property
     def available(self) -> bool:
         """Return if entity is available."""
-        return self.coordinator.last_update_success and self.coordinator.data["printer"]
+        return (
+            self.coordinator.last_update_success
+            and self.coordinator.data["printer"]
+            and self.native_value
```

Comment:
Same as above.

Reply:
Does a None value mean that the device supplying that measurement is offline, or does it mean we don't know?

Reply:
None means don't know/not supported in the current state.

Reply:
If we don't know that the device feature is offline, we should not set the entity as unavailable.

If we don't know the state yet, we should return `None` for state. This will be converted to unknown state by the base entity class.

If a feature is not supported we should probably not create that entity at all.

Reply:
Do we expect the values to be not None later so it's still good to add these entities?

Reply:
Yeah, based on what I am seeing in the octoprint code, it is possible have a value show up at a later point.